linera-views/tests/operations_tests.rs

```diff
@@ -326,3 +326,73 @@ async fn test_big_value_read_write() {
         assert_eq!(read_string, test_string);
     }
 }
+
+// The DynamoDb has limits at 1M (for pagination), 4M (for write)
+// Let us go go right past them at 20M of data with writing and then
+// reading it. And 20M is not huge by any mean. All KeyValueStoreClient
+// must handle that.
+//
+// The size of the value vary as each size as its own issues.
+#[cfg(test)]
+async fn run_big_write_read<OP: KeyValueStoreClient + Sync>(key_value_operation: OP) {
+    let mut rng = rand::rngs::StdRng::seed_from_u64(2);
+    let target_size = 20000000;
+    for (pos, value_size) in [100, 1000, 200000, 5000000].into_iter().enumerate() {
+        let n_entry = target_size / value_size;
+        let mut batch = Batch::new();
+        let key_prefix = vec![0, pos as u8];
+        let mut key_values1 = BTreeMap::new();
+        for i in 0..n_entry {
+            let mut key = key_prefix.clone();
+            bcs::serialize_into(&mut key, &i).unwrap();
+            //
+            let value = get_random_byte_vector(&mut rng, &[], value_size);
+            key_values1.insert(key.clone(), value.clone());
+            batch.put_key_value_bytes(key, value);
+        }
+        key_value_operation.write_batch(batch, &[]).await.unwrap();
+        // Checking the consistency
+        let mut key_values2 = BTreeMap::new();
+        for key_value in key_value_operation
+            .find_key_values_by_prefix(&key_prefix)
+            .await
+            .unwrap()
+            .iterator()
+        {
+            let key_value = key_value.unwrap();
+            let mut key = key_prefix.clone();
+            key.extend(key_value.0);
+            key_values2.insert(key, key_value.1.to_vec());
+        }
+        assert_eq!(key_values1, key_values2);
+    }
+}
+
+#[tokio::test]
+async fn test_big_write_read_memory() {
+    let key_value_operation = create_memory_client();
+    run_big_write_read(key_value_operation).await;
+}
+
+// TODO(#1092): That test fails for value of 5M, probably needs value splitting.
```

Comment:
What is the exact error message? It could fail for another reason. This page implies that there is no actual limit and that the recommended maximum sizes in practice are 8MB for keys and 3GB for values:
https://github.com/aayushKumarJarvis/rocks-wiki/blob/master/RocksDB-FAQ.md

Reply:
Also this https://github.com/facebook/rocksdb/issues/513

Reply:
Yes, there is a problem and I will investigate it later.